microsoft/presidio
# Quick local play using KIND

**Tl;dr**
Clone the repo, Navigate to the 'deployment' directory and run './run-with-kind.sh'.

**Details:**
KIND's prerequisites are Docker (must) and Go (Optional).
By actually running make build we can disregard the Go prerequisites (Go will be fetched while building).

**How to test:**
Option 1 - Mac / Linux users. run as documented.

Option 2 - Windows / Mac / Linux can follow these steps:

0. Create an Ubuntu VM (make sure you have at least 4 CPUs, or the analyzer won't start and make sure the machine as at least 8 GB ram).

1. install make:

```sh
sudo apt install make
```

2. Install [docker](https://docs.docker.com/install/linux/docker-ce/ubuntu/)
```sh
sudo apt-get update

sudo apt-get install \
    apt-transport-https \
    ca-certificates \
    curl \
    gnupg-agent \
    software-properties-common

curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -

sudo add-apt-repository \
   "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
   $(lsb_release -cs) \
   stable"

sudo apt-get update

sudo apt-get install docker-ce docker-ce-cli containerd.io

```
3. Verify Docker was installed :

```sh
sudo docker run hello-world
```

4. Install kubectl:

```sh
curl -LO https://storage.googleapis.com/kubernetes-release/release/`curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt`/bin/linux/amd64/kubectl

chmod +x ./kubectl

sudo mv ./kubectl /usr/local/bin/kubectl

kubectl version --client
```

5. Install Helm:
```sh
curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3

chmod 700 get_helm.sh

./get_helm.sh
```

Now you have Docker, kubectl and Helm.
6. Install Presidio:
```sh
cd deployment
./run-with-kind.sh
```
Wait for Presidio to finish starting up (all pods running)
port forward to the api micro service, give it a try
for example:

**create template:**

```sh
curl -vL -H "Content-Type: application/json" -X POST -d '{"fields": [{ "name": "PASSPORT"}, { "name": "PERSON" }]}' http://localhost:8080/api/v1/templates/123/analyze/1234
```

**Analyze:**

```sh
curl -vL -H "Content-Type: application/json" -X POST -d '{ "text": "Hi John, here is X4316545 please make update it today in Israel","analyzeTemplateId": "1234"}' http://localhost:8080/api/v1/projects/123/analyze
```
